### PR TITLE
feat: write block architecture for plugins

### PR DESCRIPTION
**Pull request by nicholaschuayunzhi**
Related #25 
Currently, behaviors can be plugged into various blocks using observer pattern.
As developer you just have to write your plugin for your behaviour.

# The Plan
A plugin will receive callbacks (via a delegate system) from `BlockBehaviour` and will respond accordingly. 
Currently implemented:
`OnFaceClick(BlockFaceBehaviour)`
Called when a face of an interact-able block is clicked.

# TODOS:
Interfaces for plugins, faces, blocks so that a plugin can't access variables that may affect the operations of other plugins.

Let me know if you see any potential problems, or you don't understand this architecture.

### CODE REVIEW

**Comment by nicholaschuayunzhi** [comment]
There is regression in the code that reintroduces the block merging behaviour #3.

EDIT: 
Identified that the regression came from https://github.com/this-place/this.place/pull/23
Sorry my bad. Will work on the fix now.

File: Assets/Scripts/BlockBehaviour.cs [hunk 1]
```diff
@@ -15,35 +15,68 @@ public class BlockBehaviour : MonoBehaviour
...
+        }
     }
 
-    // Update is called once per frame
     private void Update()
     {
         if (_isTranslating)
         {
-           TranslateBlock();
+            TranslateBlock();
```
Comment (Rinder5) [inline Assets/Scripts/BlockBehaviour.cs:75]: was this extra white space intended? =P Also, I noticed that a bunch of the code in this pr is similar to the code in #28. Does this remove the need for that pr?

Comment (nicholaschuayunzhi) [reply Assets/Scripts/BlockBehaviour.cs:75]: Ooh thanks for pointing out.

> Also, I noticed that a bunch of the code in this pr is similar to the code in #28. Does this remove the need for that pr?

I'm planning to rebase after merging the other pr

Comment (nicholaschuayunzhi) [reply Assets/Scripts/BlockBehaviour.cs:75]: Hmm seems like it is intended. The formatting was off prior to this change

**Pull request merged**